repos/system_upgrade/common/actors/checksaphana/actor.py

```diff
@@ -12,17 +14,51 @@ class CheckSapHana(Actor):
     If the upgrade flavour is 'default' no checks are being executed.
 
     The following checks are executed:
-    - If this system is _NOT_ running on x86_64, the upgrade is inhibited.
-    - If SAP HANA 1 has been detected on the system the upgrade is inhibited since it is not supported on RHEL8.
+    - If the major target release is 8, and this system is _NOT_ running on x86_64, the upgrade is inhibited.
+    - If the major target release is 9, and this system is _NOT_ running on x86_64 or ppc64le,
+      the upgrade is inhibited.
+    - If SAP HANA 1 has been detected on the system the upgrade is inhibited since there is no supported upgrade path
+      with installed SAP HANA 1.
     - If SAP HANA 2 has been detected, the upgrade will be inhibited if an unsupported version for the target release
-      has been detected.
+      has been detected (<8.8, <9.2).
+    - If the target release >=8.8 or >=9.2, the upgrade will be inhibited unless a user confirms to proceed
+      for the currently installed SAP HANA 2.0 version and the chosen target release.
     - If SAP HANA is running the upgrade is inhibited.
     """
 
     name = 'check_sap_hana'
     consumes = (SapHanaInfo,)
     produces = (Report,)
     tags = (IPUWorkflowTag, ChecksPhaseTag)
+    dialogs = (
+        Dialog(
+            scope='check_sap_hana',
```

Comment:
@alexxa just thinking, the current solution will also mean that if users answer True for IPU 7 -> 8, it will be already answered also for 8 -> 9! Even when the required version for the target system will be different. Thinking out loud, possibly you want to create scope `check_sap_hana_el8` and `check_sap_hana_el9` - it would mean you would need to create 2 different dialogs, but it would prevent a problem of automatically answered question. Later it could be possibly changed to a generator like:
```python
    dialogs =(for i in _gen_dialogs())

...
    @staticmethod
    def _gen_dialogs():
        for major_release in ('8', '9'):
            yield Dialog(scope='check_sap_hana_el{}'.format(major_release),
                ...
```
The problem is that we used dialogs for specific cases that have been connected always to a specific upgrade path, but now we introduces a requirement for a dialog(s) for multiple upgrade paths, which is different from previous scenearious.

@vinzenz @oamg/developers  any thoughts on this?

Reply:
Petr, I don't follow why it's valid for 7to8 case when get_saphana_answer and dialog inside are called only if a target release >8.6 and >9.0:
https://github.com/oamg/leapp-repository/pull/1077/files/84b14e4a20b26af380963927b7e0d1a323f470cf#diff-388596db3aa9f19b3e3ba62b12d22a70c8af01075f0384a4526cbbd4326691b0R198

Reply:
@alexxa the upgrade from rhel 7.9 -> 8.8 means the target is 8.8 which triggers the question. user will answer it and it's answered. if they later upgrade 8.8 -> 9.2, they will not be asked anymore because they answered the questions already for the upgrade 7.9 -> 8.8. and we can continue with the upgrade 9.6 -> 10.0 with the same problem.

Reply:
ah, do you mean inside for loop for multiple instances if answered once? =(((( but anyway, even if there are multiple HANA instances, they belong to one RHEL and one upgrade path, right?

Reply:
Discussed via video call. We have decided to get rid of the user question as it's more feasible solution and keep it on the report only. As this is not real blocker for the upgrades and users are quired to read the report & documentation. Currently we have already some reports explaining people they are supposed to do some manual steps before the upgrade and they are not inhibitors neither.

@oamg/developers anyway, I think we have a nice topic for the discussion about dialogs. I will create a discussion topic later for that.